templates/coredns/coredns.py

```diff
@@ -0,0 +1,119 @@
+from jumpscale import j
+from zerorobot.template.base import TemplateBase
+
+
+ETCD_TEMPLATE_UID = 'github.com/threefoldtech/0-templates/etcd/0.0.1'
+
+NODE_CLIENT = 'local'
+
+class Coredns(TemplateBase):
+    version = '0.0.1'
+    template_name = 'coredns'
+
+    def __init__(self, name, guid=None, data=None):
+        super().__init__(name=name, guid=guid, data=data)
+        self._node_ = None
+        self.add_delete_callback(self.uninstall)
+        self.recurring_action('_monitor', 30)  # every 30 seconds
+
+    def _monitor(self):
+        self.logger.info('Monitor coredns %s' % self.name)
+        self.state.check('actions', 'start', 'ok')
+
+        if not self._coredns_sal.is_running():
+            self.state.delete('status', 'running')
+            self._coredns_sal.deploy()
+            self._coredns_sal.start()
+            if self._coredns_sal.is_running():
+                self.state.set('status', 'running', 'ok')
+        else:
+            self.state.set('status', 'running', 'ok')
+    @property
+    def _node_sal(self):
+        """
+        connection to the node
+        """
+        return j.clients.zos.get(NODE_CLIENT)
+
+    @property
+    def _node(self):
+        if not self._node_:
+            self._node_ = self.api.services.get(template_account='threefoldtech', template_name='node')
+        return self._node_
+
+    @property
+    def _coredns_sal(self):
+        kwargs = {
+            'name': self.name,
+            'node': self._node_sal,
+            'recursive_resolvers':self.data['upsteram'],
+            'zt_identity': self.data['ztIdentity'],
+            'nics': self.data['nics'],
+            'etcd_endpoint': self._etc_url
+        }
+        return j.sal_zos.coredns.get(**kwargs)
+    @property
+    def _etcd(self):
+        return self.api.services.get(template_uid=ETCD_TEMPLATE_UID, name=self.data['etcdServerName'])
+    @property
+    def _etc_url(self):
+        result = self._etcd.schedule_action('connection_info').wait(die=True).result
+        return result['client_url']
+        
+    def node_port(self):
+        return self.data['nodePort']
```

Comment:
this doesn't exist in the service schema.
also you should return the value from the coredns_sal